prymer/primer3/primer3.py

```diff
@@ -534,51 +525,44 @@ def _build_primers(
                 bases = reverse_complement(bases)
 
             primers.append(
-                Primer(
+                Oligo(
                     bases=bases,
                     tm=float(design_results[f"PRIMER_{design_task.task_type}_{idx}_TM"]),
                     penalty=float(design_results[f"PRIMER_{design_task.task_type}_{idx}_PENALTY"]),
                     span=span,
+                    self_any_th=float(design_results[f"{key}_SELF_ANY_TH"]),
+                    self_end_th=float(design_results[f"{key}_SELF_END_TH"]),
+                    hairpin_th=float(design_results[f"{key}_HAIRPIN_TH"]),
                 )
             )
         return primers
 
     @staticmethod
-    def _assemble_primers(
-        design_input: Primer3Input, design_results: dict[str, str], unfiltered_designs: list[Primer]
+    def _assemble_single_designs(
+        design_input: Primer3Input,
+        design_results: dict[str, str],
+        unfiltered_designs: list[Oligo],
     ) -> Primer3Result:
-        """Helper function to organize primer designs into valid and failed designs.
-
-        Wraps `Primer3._is_valid_primer()` and `Primer3._build_failures()` to filter out designs
-        with dinucleotide runs that are too long and extract additional failure reasons emitted by
-        Primer3.
-
-        Args:
-            design_input: encapsulates the target region, design task, specifications,
-             and scoring penalties
-            unfiltered_designs: list of primers emitted from Primer3
-             design_results: key-value pairs of results reported by Primer3
+        """Screens oligo designs (primers or probes) emitted by Primer3 for acceptable dinucleotide
+        runs and extracts failure reasons for failed designs."""
 
-        Returns:
-            primer_designs: a `Primer3Result` that encapsulates valid and failed designs
-        """
-        valid_primer_designs = [
+        valid_oligo_designs = [
             design
             for design in unfiltered_designs
-            if Primer3._is_valid_primer(primer_design=design, design_input=design_input)
+            if _has_acceptable_dinuc_run(oligo_design=design, design_input=design_input)
         ]
         dinuc_failures = [
             design
             for design in unfiltered_designs
-            if not Primer3._is_valid_primer(primer_design=design, design_input=design_input)
+            if not _has_acceptable_dinuc_run(oligo_design=design, design_input=design_input)
         ]
 
         failure_strings = [design_results[f"PRIMER_{design_input.task.task_type}_EXPLAIN"]]
         failures = Primer3._build_failures(dinuc_failures, failure_strings)
-        primer_designs: Primer3Result = Primer3Result(
-            filtered_designs=valid_primer_designs, failures=failures
+        design_candidates: Primer3Result = Primer3Result(
+            filtered_designs=valid_oligo_designs, failures=failures
```

Comment:
This came up elsewhere in conversation that perhaps `filtered_designs` should just be `designs`?  If `primer3` implemented a dinuc check, would we feel the need to call this `filtered_designs`?

Reply:
I have updated the docstring and changed the variable from `filtered_designs` to `designs`. These are filtered for out-of-spec characteristics and the corresponding `failures` list contains the reasons why (so I have been thinking of these designs as "filtered" because they have undergone one round of filtering for primer-specific properties like GC content, TM, etc.).